### PR TITLE
Fixed crash with declaring typed variable of undefined type

### PR DESCRIPTION
**Pull request by aparajit-pratap**
This fixes defect: http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6607

### CODE REVIEW

**Comment by aparajit-pratap** [comment]
@junmendoza PTAL

File: test/Engine/ProtoTest/Associative/MicroFeatureTests.cs [hunk 1]
```diff
@@ -3169,6 +3169,21 @@ public void TestNullsOnExpression01()
...
+        b : UndefinedType2 = 2;
+        c : UndefinedType3 = null;
+";
+            thisTest.RunScriptSource(code);
+            thisTest.Verify("a", null);
+            thisTest.Verify("b", null);
+            thisTest.Verify("c", null);
 
+            thisTest.VerifyRuntimeWarningCount(3);
+        }
```
Comment (junmendoza) [inline test/Engine/ProtoTest/Associative/MicroFeatureTests.cs]: git diff sucks.
Can you just comment here the line of code you added in the testcase :)

Comment (aparajit-pratap) [inline test/Engine/ProtoTest/Associative/MicroFeatureTests.cs]:
```
[Test]
        public void TestUndefinedTypedIdentifier()
        {
            string code =
@"
        a : UndefinedType;
        b : UndefinedType2 = 2;
        c : UndefinedType3 = null;
";
            thisTest.RunScriptSource(code);
            thisTest.Verify("a", null);
            thisTest.Verify("b", null);
            thisTest.Verify("c", null);

            thisTest.VerifyRuntimeWarningCount(3);
        }
```

**Comment by junmendoza** [comment]
Looks good.
LGTM if no regressions

File: test/Engine/ProtoTest/Associative/MicroFeatureTests.cs (hunk 1 above)
Comment (aparajit-pratap) [inline test/Engine/ProtoTest/Associative/MicroFeatureTests.cs]: Probably Git cannot handle large file diffs

**Comment by aparajit-pratap** [comment]
No regressions in `ProtoTest`. Merging ...

**Pull request merged**